go/plugins/googlegenai/gemini.go

```diff
@@ -69,6 +73,152 @@ type EmbedOptions struct {
+	// The harm category is harassment.
+	HarmCategoryHarassment HarmCategory = "HARM_CATEGORY_HARASSMENT"
+	// The harm category is sexually explicit content.
+	HarmCategorySexuallyExplicit HarmCategory = "HARM_CATEGORY_SEXUALLY_EXPLICIT"
+	// The harm category is civic integrity.
+	HarmCategoryCivicIntegrity HarmCategory = "HARM_CATEGORY_CIVIC_INTEGRITY"
+)
+
+// Specify if the threshold is used for probability or severity score. If not specified,
+// the threshold is used for probability score.
+type HarmBlockMethod string
+
+const (
+	// The harm block method is unspecified.
+	HarmBlockMethodUnspecified HarmBlockMethod = "HARM_BLOCK_METHOD_UNSPECIFIED"
+	// The harm block method uses both probability and severity scores.
+	HarmBlockMethodSeverity HarmBlockMethod = "SEVERITY"
+	// The harm block method uses the probability score.
+	HarmBlockMethodProbability HarmBlockMethod = "PROBABILITY"
+)
+
+// The harm block threshold.
+type HarmBlockThreshold string
+
+const (
+	// Unspecified harm block threshold.
+	HarmBlockThresholdUnspecified HarmBlockThreshold = "HARM_BLOCK_THRESHOLD_UNSPECIFIED"
+	// Block low threshold and above (i.e. block more).
+	HarmBlockThresholdBlockLowAndAbove HarmBlockThreshold = "BLOCK_LOW_AND_ABOVE"
+	// Block medium threshold and above.
+	HarmBlockThresholdBlockMediumAndAbove HarmBlockThreshold = "BLOCK_MEDIUM_AND_ABOVE"
+	// Block only high threshold (i.e. block less).
+	HarmBlockThresholdBlockOnlyHigh HarmBlockThreshold = "BLOCK_ONLY_HIGH"
+	// Block none.
+	HarmBlockThresholdBlockNone HarmBlockThreshold = "BLOCK_NONE"
+	// Turn off the safety filter.
+	HarmBlockThresholdOff HarmBlockThreshold = "OFF"
+)
+
+// Safety settings.
+type SafetySetting struct {
+	// Determines if the harm block method uses probability or probability
+	// and severity scores.
+	Method HarmBlockMethod `json:"method,omitempty"`
+	// Required. Harm category.
+	Category HarmCategory `json:"category,omitempty"`
+	// Required. The harm block threshold.
+	Threshold HarmBlockThreshold `json:"threshold,omitempty"`
+}
+
+// GeminiConfig mirrors GenerateContentConfig without direct genai dependency
+type GeminiConfig struct {
+	ai.GenerationCommonConfig
+
+	// Safety settings
+	SafetySettings []*SafetySetting `json:"safetySettings,omitempty"`
+
+	// Model version
+	Version string `json:"version,omitempty"`
+}
+
+// extractConfigFromInput converts any supported config type to GoogleAIConfig
+func extractConfigFromInput(input *ai.ModelRequest) (*GeminiConfig, error) {
+	var result GeminiConfig
+	switch config := input.Config.(type) {
+	case GeminiConfig:
+		return &config, nil
+	case *GeminiConfig:
+		return config, nil
+	case ai.GenerationCommonConfig:
+		return convertFromCommonConfig(config), nil
+	case *ai.GenerationCommonConfig:
+		if config == nil {
+			return &result, nil
+		}
+		return convertFromCommonConfig(*config), nil
+	case map[string]any:
+		log.Printf("Warning: Using map[string]any for config may silently ignore unknown parameters")
```

Comment:
I was imagining we would check for unknown fields here and warn only then but that might be complex. Let's add the TODO back in and get rid of this -- there's nothing wrong with using a map *if* there are no unknown fields.